R-packages/covidcast/tests/testthat/test-covidcast.R

```diff
@@ -104,8 +105,51 @@ with_mock_api({
     )
   })
 
+  test_that("covidcast_signal works across multiple days with gaps", {
+    # If we request 3 days, we'll get 3 API queries. If the middle day is
+    # missing, we should get an appropriate warning, but still get the right
+    # data frame.
+
+    # day 1: api.php-32641f.csv
+    # day 2: api.php-b85d44.csv (empty)
+    # day 3: api.php-f49e8f.csv
+    expect_warning(covidcast_signal("foo", "bar", "2020-01-10", "2020-01-12",
+                                    geo_type = "county"),
+                   class = "covidcast_fetch_failed")
+
+    res <- suppressWarnings(
+      covidcast_signal("foo", "bar", "2020-01-10", "2020-01-12",
+                       geo_type = "county"))
+    expect_equal(
+      res,
+      structure(data.frame(
+        data_source = "foo",
+        signal = "bar",
+        geo_value = c("01001", "01002", "31001", "31002"),
+        time_value = as.Date(c("2020-01-10", "2020-01-10",
+                               "2020-01-12", "2020-01-12")),
+        issue = as.Date(c("2020-01-11", "2020-01-11",
+                          "2020-01-13", "2020-01-13")),
+        lag = 1,
+        value = c(91.2, 99.1, 81.2, 89.1),
+        stderr = c(0.8, 0.2, 0.8, 0.2),
+        sample_size = c(114.2, 217.8, 314.2, 417.8)),
+        class = c("covidcast_signal", "data.frame"),
+        metadata = structure(data.frame(
+          data_source = "foo",
+          signal = "bar",
+          time_type = "day",
+          geo_type = "county",
+          min_time = as.Date("2020-01-01"),
+          max_time = as.Date("2020-01-02"),
+          min_value = 0,
+          max_value = 10,
+          max_issue = as.Date("2020-04-04")),
+          class = c("covidcast_meta", "data.frame"))))
+  })
+
   test_that("covidcast_signal stops when end_day < start_day", {
-    # reusing api.php-da6974.json
+    # reusing api.php-d2e163.json for metadata
```

Comment:
Is this still json?

Reply:
Good catch, and the filename was wrong too (the tooling for httptest filename tracking is pretty clunky)